internal/operator/cluster/clusterlogic.go

```diff
@@ -804,22 +799,17 @@ func waitForDeploymentReady(clientset kubernetes.Interface, namespace, deploymen
 	ctx := context.TODO()
 
 	// set up the timer and timeout
-	// first, ensure that there is an available Pod
-	timeout := time.After(timeoutSecs)
-	tick := time.NewTicker(periodSecs)
-	defer tick.Stop()
-
-	for {
-		select {
-		case <-timeout:
-			return fmt.Errorf("readiness timeout reached for deployment %q", deploymentName)
-		case <-tick.C:
-			// check to see if the deployment is ready
-			if d, err := clientset.AppsV1().Deployments(namespace).Get(ctx, deploymentName, metav1.GetOptions{}); err != nil {
-				log.Warn(err)
-			} else if d.Status.Replicas == d.Status.ReadyReplicas {
-				return nil
-			}
+	if err := wait.Poll(periodSecs, timeoutSecs, func() (bool, error) {
+		// check to see if the deployment is ready
+		d, err := clientset.AppsV1().Deployments(namespace).Get(ctx, deploymentName, metav1.GetOptions{})
+		if err != nil {
+			log.Warn(err)
 		}
+
+		return d.Status.Replicas == d.Status.ReadyReplicas, nil
```

Comment:
This used to loop when `err != nil`. When there's an error, `d` is probably empty which would return true:
`0 == d.Status.Replicas == d.Status.ReadyReplicas`.

```suggestion
		return err == nil && d.Status.Replicas == d.Status.ReadyReplicas, nil
```

Reply:
Ah good catch. That would panic otherwise.